tests/v2/integration/steveapi/steve_api_test.go

```diff
@@ -336,6 +337,42 @@ func (s *LocalSteveAPITestSuite) TestExtensionAPIServerAuthorization() {
 	}
 }
 
+func (s *LocalSteveAPITestSuite) TestExtensionAPIServerPostRequests() {
+	client, err := rest.HTTPClientFor(s.client.WranglerContext.RESTConfig)
+	require.NoError(s.T(), err)
+
+	tests := []struct {
+		name string
+		path string
+		body io.Reader
+
+		// expectedBody needs to be a regex to accound for fields in the received body that cannot be reliably dtermined (such as "time" and "manager")
+		expectedBody *regexp.Regexp
+		expectedCode int
+	}{
+		{
+			name: "post to selfuser",
+			path: "v1/ext.cattle.io.selfuser",
+			body: strings.NewReader(`{"kind":"selfuser"}`),
+
+			expectedBody: regexp.MustCompile(`{"type":"ext.cattle.io.selfuser","links":{"view":"https://` + s.client.WranglerContext.RESTConfig.Host + `/apis/ext.cattle.io/v1/selfusers"},"apiVersion":"ext.cattle.io/v1","kind":"SelfUser","metadata":{"generateName":"e-","managedFields":\[{"apiVersion":"ext.cattle.io/v1","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:generateName":{}}},"manager":".*","operation":"Update","time":".*"}],"relationships":null,"state":{"error":false,"message":"Resource is current","name":"active","transitioning":false}},"status":{"userID":"user-[a-z0-9]+"}}`),
```

Comment:
Maybe it's good enough to check for status field (with a typed Go struct) instead of a hard to maintain regex?